inst/rmarkdown/templates/isba/resources/template.tex

```diff
@@ -0,0 +1,163 @@
+  {$if(csl-hanging-indent)$\setlength{\parindent}{0pt}%
+  \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces$endif$}%
+  {\par}
+% For Pandoc 2.11+
+\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
+ {% don't indent paragraphs
+  \setlength{\parindent}{0pt}
+  % turn on hanging indent if param 1 is 1
+  \ifodd #1
+  \let\oldpar\par
+  \def\par{\hangindent=\cslhangindent\oldpar}
+  \fi
+  % set entry spacing
+  \setlength{\parskip}{#2\cslentryspacingunit}
+ }%
+ {}
+\usepackage{calc}
+\newcommand{\CSLBlock}[1]{#1\hfill\break}
+\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
+\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
+\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
+$endif$
+
+% Pandoc header
+$for(header-includes)$
+$header-includes$
+$endfor$
+
+$preamble$
+
+\begin{document}
+\begin{frontmatter}
+
+\title{$title$}
+\runtitle{$title$}
+
+\begin{aug}
+$for(author/pairs)$
+\author{$author.value.firstname$ $author.value.lastname$\thanksref{$author.value.affiliationref$$if(author.value.footnoterefs)$,$for(author.value.footnoterefs)$$it$$sep$,$endfor$$endif$}$if(author.value.email)$\ead[label=ea-$author.key$,email]{$author.value.email$}$endif$$if(author.value.url)$\ead[label=ua-$author.key$,url]{$author.value.url$}$endif$}$sep$, $endfor$
+
+\runauthor{$for(author/first)$$it.lastname$$endfor$$if(author/rest/rest)$ et al.$else$$for(author/rest)$ \and $author.lastname$$endfor$$endif$}
+
+$for(affiliations)$ %loop through affiliations
+\address[$affiliations.ref$]{$affiliations.name$, $affiliations.address$ $for(author/pairs)$
+$if(it.value.email)$
+\ifstrequal{$affiliations.ref$}{$it.value.affiliationref$}{\printead{ea-$it.key$}}{}
+$endif$
+$if(it.value.url)$
+\ifstrequal{$affiliations.ref$}{$it.value.affiliationref$}{\printead{ua-$it.key$}}{}
+$endif$
+$sep$ $endfor$%authors
+}$endfor$%affiliations
+
+$for(footnotes)$
+\thankstext{$it.ref$}{$it.text$}
+$endfor$
+
+\end{aug}
+
+\begin{abstract}
+$abstract$
+\end{abstract}
+
+% MSC class
+$if(MSC2020primary)$ \begin{keyword}[class=MSC]
+$for(MSC2020primary/first)$\kwd[Primary ]{$it$}$endfor$
+$if(MSC2020primary/rest)$$for(MSC2020primary/rest)$\kwd{$it$}$sep $$endfor$$endif$
+$if(MSC2020secondary)$$for(MSC2020secondary/first)$\kwd[; secondary ]{$it$}$endfor$
+$if(MSC2020secondary/rest)$$for(MSC2020secondary/rest)$\kwd{$it$}$sep $$endfor$$endif$$endif$\end{keyword}$endif$
+% keywords
+$if(keywords)$ \begin{keyword} $for(keywords)$\kwd{$keywords$}$sep$ $endfor$\end{keyword}$endif$
+\end{frontmatter}
+
+$for(include-before)$
+$include-before$
+$endfor$
+
+$body$
+
+$for(include-after)$
+$include-after$
+$endfor$
+
+$if(supplements)$
+$for(supplements)$
+\begin{supplement}
+\stitle{$it.title$}
+\sdescription{$it.description$}
+\end{supplement}
+$endfor$$endif$
+
+$if(acknowledgements)$
+\begin{acks}[Acknowledgments]
+$acknowledgements$
+\end{acks}
+$endif$
+
+\bibliographystyle{ba}
+\bibliography{$bibliography$}
```

Comment:
Seems like only `natbib` is supported as those command are not behind a `if`. Is it the case ? 

Can you use with Pandoc citeproc. I see a `csl` field used in the skeleton, but it should work only when `citatation_package: "default"` (which will be pandoc's) and not with natbib. 

Should `natbib` been enforced like in `informs_article()` ? 

Pandoc citeproc have some benefit so it would be good if this is supported too.

Reply:
I clearly don't know enough to have a say about alternative referencing systems. Natbib is something that is enforced in the template. I am happy to enforce it as `informs_article()` does, but I am open to trying alternative suggestions.

Reply:
Having an alternative using Pandoc citeproc and a CSL  would mean : 

- There is a CSL file that can be used to have the same expected style 
- `$if(natbib)$` should be added in the template around the part that does not need to be included if `natbib` is not used
- `citation_package: "default"` in the format function would trigger Pandoc citeproc to create the references sections below your last header with id `#references`

I did not find a CSL for ISBA in https://www.zotero.org/styles but I may not know the correct name. `ba.bst` is probably renamed from another style, maybe AMS one. 

Anyway, I am fine doing like `informs_article()` if we don't have the right option for this journal. 

Do you know if a `isba.csl` or `ba.csl` is available ?

Reply:
No and cant find any. I will enforce `natbib` then.